core/src/main/java/com/scalar/db/storage/cassandra/MutateStatementHandler.java

```diff
@@ -55,8 +53,7 @@ public ResultSet handle(Operation operation) throws ExecutionException {
           throw new ReadRepairableExecutionException(
```

Comment:
This is thrown when the write type is SIMPLE and the mutation has conditions. But, I think it's possible the cause of the failure is not related to the condition, and throwing `ExecutionException` might be safer? Sorry, I don't correctly understand what ReadRepairableExecutionException means and how to handle it, and I might be missing something.

Reply:
Ah, sorry. I was confused... Thanks for pointing it out.

> The Cassandra adaptor throws this exception when the error can be repaired by a following read.

You expect the client will re-read the record and decide the next operation, when catching the exception? If so, it sounds to be similar to the error handling of `ExecutionException` to me. Is there any significant difference?

BTW, I couldn't find any documentation and guidance for `ReadRepairableExecutionException`, so I guess the users don't know what to do when catching the exception...